lib/src/charts/painters/cartesian/bar_painter.dart

```diff
@@ -24,17 +23,83 @@ class BarPainter implements CartesianPainter {
   late Paint _barPaint;
   late Paint _barStroke;
 
-  BarPainter({
-    required this.useGraphUnits,
-  }) {
+  BarPainter({required this.useGraphUnits}) {
     _barPaint = Paint();
     _barStroke = Paint()..style = PaintingStyle.stroke;
   }
 
+  _BarInteractionData _findNearestBar(
+    Offset position,
+    List<_BarInteractionData> bars, {
+    double fuzzinessRange = 0.0,
+  }) {
+    return bars.reduce((previous, current) {
+      final distToPrev = position.dx - previous.rect.right;
+      final distToCurrent = current.rect.left - position.dx;
+
+      return distToPrev >= distToCurrent ? current : previous;
+    });
+  }
+
+  @override
+  BarInteractionResult? hitTest(
+    TouchInteractionType type,
+    Offset localPosition,
+  ) {
+    // We will perform HitTest only if Interactions are enabled for this series.
+    if (_data.interactionEvents.isEnabled) {
+      var snapToNearestBar = _data.interactionEvents.snapToNearestPoint;
+      var fuzzinessRange = _data.interactionEvents.snappingRange;
```

Comment:
Let's rename this. This should be named `fuzziness` everywhere.
Also calling this range doesn't make sense since this is just a number that you add to existing values.
So just calling this `fuzziness` should be fine. 
About this behind easy to understand for the user, whatever we name it, it's going to need some explanation in docs to make it clear.